src/main/java/uk/gov/hmcts/reform/pip/account/management/service/AccountService.java

```diff
@@ -148,10 +149,36 @@ public Map<CreationEnum, List<?>> addUsers(List<PiUser> users, String issuerEmai
                 continue;
             }
 
-            PiUser addedUser = userRepository.save(user);
-            createdAccounts.add(addedUser.getUserId());
+            try {
```

Comment:
I havn't tried it, but if the user already exists in Azure (which the frontend will try to create first when you click 'Submit'), and therefore that request fails, will it get to this part of the code?

If it does then all fine :) However if not we have an issue where it will fail at that bit then never reach this bit of code

Reply:
This section of code adding user in pi_user table. addAzureAccounts that's the function which is trying to add user into  azure AAD.

Reply:
@ChrisS1512 the FE will use the response of the first request to fire the second request so it should not get to this part of the code, IIRC, FE takes the accounts from CREATED_ACCOUNTS to build the 2nd request to add to PI db, so if 1st request failed and successfully adds them to ERRORED_ACCOUNTS you shouldnt get here